ary/models.py

```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
+            if schema['type'] == Field.MULTISELECT:
+                value = [schema['options'].get(x, x) for x in value]
+            return {schema['name']: value}
+        if isinstance(schema, dict):
+            data = {
+                k: self._data_from_schema(v, raw_data)
+                for k, v in schema.items()}
+        elif isinstance(schema, list):
+            data = [self._data_from_schema(x, raw_data) for x in schema]
+        else:
+            raise Exception("Something that could not be parsed from schema")
+        return data
+
+    def get_metadata_json(self):
+        metadata_schema = self._load_schema(MetadataField, MetadataGroup)
+        metadata_raw = self.metadata or {}
+        metadata_raw = metadata_raw.get('basic_information', {})
+        metadata = self._data_from_schema(metadata_schema, metadata_raw)
+        return metadata
+
+    def get_methodology_json(self):
+        methodology_sch = self._load_schema(MethodologyField, MethodologyGroup)
+        methodology_raw = self.methodology or {}
+
+        mapping = {
+            'attributes': lambda x: self._data_from_schema(methodology_sch, x),
+            'sectors': lambda x: Sector.objects.get(id=x).title,
+            'focuses': lambda x: Focus.objects.get(id=x).title,
+            'affected_groups': lambda x: x['name'],
+            'locations': lambda x: GeoArea.objects.get(id=x).title,
+            'objectives': identity,
+            'sampling': identity,
+            'limitations': identity,
+            'data_collection_techniques': identity
+        }
+
+        return {
+            underscore_to_title(k):
+                v if not isinstance(v, list)
+                else [mapping[k](y) for y in v]
+            for k, v in methodology_raw.items()
+        }
+
+    def get_summary_json(self):
+        # functions to get exact value of an entry in summary group
+        def identity(x):
+            return x
+        value_functions = {
+            'specific_need_group': lambda x: SpecificNeedGroup.objects.get(
+                id=x).title,
+            'affected_group': lambda x: AffectedGroup.objects.get(id=x).title,
+            'underlying_factors': identity,
+            'outcomes': identity,
+            'affected_location': lambda x: AffectedLocation.objects.get(
+                id=x).title,
+            'priority_sector': lambda x: PrioritySector.objects.get(
+                id=x).title,
+            'priority_issue': lambda x: PriorityIssue.objects.get(id=x).title
+        }
+
+        # Formatting of underscored keywords, by default is upper case as given
+        # by default_format() function below
+        formatting = {
+            'priority_sector': lambda x: 'Most Unmet Needs Sectors',
+            'affected_location': lambda x: 'Settings Facing Most Humanitarian Issues'  # noqa
+        }
+
+        default_format = underscore_to_title   # function
+
+        def default_format(x):
```

Comment:
Useless function definition as `underscote_to_title` is already defined.

Reply:
Copy That.